### PR TITLE
Validate exam profile

### LINKED ISSUE #2339: Exams: validate profile before syncing to Pearson
To cover profiles that already exist in production, we should validate the data before we send to Pearson and if they're invalid mark them as such. We probably want to at least display a message on the dashboard and possibly email them.

Particularly, all the following fields need to be filled out and fall within the CP-1252 character set:

- `first_name`
- `last_name`
- `email`
- `address`
- `city`
- `state`
- `country`
- `postal_code`
- `phone_number`

As a special case, `first_name` and/or `last_name` can be outside the CP-1252 character set, however, `romanized_first_name` and `romanized_last_name` need to have values and within the CP-1252 range.

If a profile is invalid when we go to export it, set its `ExamProfile.status` value to `ExamProfile.PROFILE_INVALID`.

### PR DESCRIPTION
**Pull request by annagav**
#### What are the relevant tickets?
Fix #2339
#### What's this PR do?
validates `ExamProfile` data before sending it to Pearson. If profile is invalid then sets status on the `ExamProfile` to 'invalid'.
#### How should this be manually tested?
```
from django.contrib.auth.models import User
user = User.objects.filter(username='username')[0]
user.profile.address=‘中华人民共和国’
user.profile.save()
```
now the ExamProfile status for this user should be set to 'pending'. Now you can the task:

```
from exams.tasks import *
result = export_exam_profiles.delay()
```
Now the ExamProfile status should be set to 'invalid', and the exported file should not contain this profile. The same should happen if any of the fields are not filled out (except for `postal_code` which is optional)

### CODE REVIEW

File: exams/utils.py [hunk 1]
```diff
@@ -72,3 +73,33 @@ def authorize_for_exam(mmtrack, course_run):
                 mmtrack.user.username,
                 course_run.edx_course_key
             )
+
+
+def match_field(profile, field):
+    """
+    If a field is filled out match it to the CP-1252 character set.
+    """
+    pattern = r'^[\u0020-\u00FF]*$'
```
Comment (noisecapella) [inline exams/utils.py]: Where does this pattern come from?

File: exams/utils.py [hunk 2]
```diff
@@ -72,3 +73,33 @@ def authorize_for_exam(mmtrack, course_run):
...
+    """
+    Make sure all the required fields fall within the CP-1252 character set
+    Args:
+        profile (Profile): user profile
+    """
+    fields = ['address', 'city', 'state_or_territory', 'country', 'phone_number']
+    optional = {'first_name': 'romanized_first_name', 'last_name': 'romanized_last_name'}
+
+    if not match_field(profile.user, 'email'):
+        return False
```
Comment (noisecapella) [inline exams/utils.py]: It looks like this line is uncovered, can you write a test for it?

File: exams/utils.py [hunk 3]
```diff
@@ -72,3 +73,33 @@ def authorize_for_exam(mmtrack, course_run):
...
+    for key, value in optional.items():
+        if not match_field(profile, key):
+            fields.append(value)
```
Comment (noisecapella) [inline exams/utils.py]: This line is uncovered too

File: exams/tasks.py [hunk 4]
```diff
@@ -46,6 +47,16 @@ def export_exam_profiles(self):
...
     file_prefix = datetime.now(pytz.utc).strftime(PEARSON_CDD_FILE_PREFIX)
 
+    valid_exam_profiles = []
+    for exam_profile in exam_profiles:
+        validated = validate_profile(exam_profile.profile)
+
+        if validated:
+            valid_exam_profiles.append(exam_profile)
+        else:
+            exam_profile.status = ExamProfile.PROFILE_INVALID
```
Comment (noisecapella) [inline exams/tasks.py]: These two lines are uncovered. Can you write a test for this case?

**Comment by annagav** [comment]
@noisecapella, I addressed all comments.

File: exams/utils.py [hunk 5]
```diff
@@ -116,3 +117,33 @@ def authorize_for_exam(mmtrack, course_run):
...
+    reg = re.compile(pattern)
+    value = getattr(profile, field)
+    return (True if reg.match(value) else False) if value else False
+
+
+def validate_profile(profile):
+    """
+    Make sure all the required fields fall within the CP-1252 character set
+    Args:
+        profile (Profile): user profile
```
Comment (noisecapella) [inline exams/utils.py]: Can you add a `Returns` section to this docstring?

File: exams/tasks_test.py [hunk 6]
```diff
@@ -143,6 +143,36 @@ def side_effect(tsv, qs):
         for exam_profile in self.expected_in_progress_profiles:
             assert exam_profile in in_progress_profiles
 
+    @patch('exams.tasks.validate_profile', return_value=False, autospec=True)
+    @patch('exams.pearson.upload.upload_tsv')
```
Comment (noisecapella) [inline exams/tasks_test.py]: Can you add `autospec=True` here too?

File: exams/utils_test.py [hunk 7]
```diff
@@ -147,3 +150,92 @@ def test_exam_authorization_when_not_passed_course(self):
...
+        ('汉字', 'Andrew', True),
+        ('', 'Andrew', True),
+        ('汉字', '', False),
+        ('', '', False)
+    )
+    @unpack
+    def test_romanized_name(self, name, romanized_name, result):
+        """
+        test romanized name optional/required
+        :return:
```
Comment (noisecapella) [inline exams/utils_test.py]: Can you remove this line?

**Pull request merged**